src/RegistrationMsg.chpl

```diff
@@ -278,6 +280,116 @@ module RegistrationMsg
         return new MsgTuple(repMsg, MsgType.NORMAL);
     }
 
+    /* 
+    Compile the component parts of a DataFrame attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg payload: name of SymTab element
+    :type payload: string
+
+    :arg argSize: number of arguments in payload
+    :type argSize: int
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachDataFrameMsg(cmd: string, payload: string, argSize: int,
+                                 st: borrowed SymTab): MsgTuple throws {        
+        var msgArgs = parseMessageArgs(payload, argSize);
+        const name = msgArgs.getValueOf("name"); 
+        var colName = "df_columns_%s".format(name);
+        var repMsg = "dataframe+%s".format(name);
+
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting DataFrame components for '%s'".format(cmd, name));
+
+        var jsonParam = msgArgs.get("name");
+        jsonParam.setVal(colName);
+        var json: [0..#1] string = [jsonParam.getJSON()];
```

Comment:
`getJSON` requires a size be provided. I would suggest updating this to 
```chapel
var keys: List(string) = new List(["name"]);
var json = msgArgs.getJSON(keys);
```

Reply:
you would still need to update the value with my previous suggestion. I would actually suggest updating to something similar to the code on 318-320.
```chapel
var colParam = new ParameterObj("name", colName, ObjectType.VALUE, "str")
var json: [0..#1] string = [colParam.getJSON()];
```

Reply:
Updated